### PR TITLE
Getting Error code from burrow execution

### PR DESCRIPTION
**Pull request by Pawlak00**
### Description of the Change
Add possibility to get info why Burrow execution failed, what is stateful validation error code

### Benefits
Ability to get error code from smart contract execution, know why it failed

### CODE REVIEW

File: goSrc/src/vmCaller/iroha/commands.go [hunk 1]
```diff
@@ -640,7 +641,8 @@ func handleErrors(result *C.Iroha_CommandError, err error, commandName string) (
 		if error_extra_ptr != nil {
 			error_extra = ": " + *error_extra_ptr
 		}
-		return fmt.Errorf("Error executing %s command: %s", commandName, error_extra)
+		IrohaErrorDetails = fmt.Sprintf("%s %s error_code %d ",commandName, error_extra, result.error_code)
```
Comment (baziorek) [inline goSrc/src/vmCaller/iroha/commands.go:644]:
```suggestion
		IrohaErrorDetails = fmt.Sprintf("%s %s error_code %d", commandName, error_extra, result.error_code)
```

Comment (baziorek) [reply goSrc/src/vmCaller/iroha/commands.go:644]: I don't know golang, but isn't `IrohaErrorDetails` global variable? If it is - is it sure that multiple smart contracts would save the value in the same time?

**Review by baziorek — COMMENTED** [review]
+ documentation changes

**Comment by iceseer** [comment]
Do we need to read ```engine_err_response``` field in a query?

File: goSrc/src/vmCaller/iroha/commands.go (hunk 1 above)
Comment (iceseer) [reply goSrc/src/vmCaller/iroha/commands.go:644]: It seems it is global. But all the fields are reinitialized in VmCall for each ```c++ -> go``` call, before the other calls made from ```go``` part.

**Comment by Pawlak00** [comment]
> Do we need to read `engine_err_response` field in a query?

There is problem when I wanted to add new field in response, because query result is implemented using boost::tuple which has max 10 elements, and result tuple can have 11 elements, so I came up with this solution

**Comment by iceseer** [comment]
@Pawlak00 It seems c++ code uses only 9 arguments from request. You can exchange unused one. It seems permissions doesn't uses https://imgur.com/3P90itj.png

**Comment by Pawlak00** [comment]
> @Pawlak00 It seems c++ code uses only 9 arguments from request. You can exchange unused one. It seems permissions doesn't uses https://imgur.com/3P90itj.png
@iceseer 
the problem is here: https://github.com/hyperledger/iroha/blob/4c4adf9be2a2374623c250e5b42cc76f40bb2700/irohad/ametsuchi/impl/postgres_specific_query_executor.cpp#L341 becouse QueryTuple and PermissionTuple are concatated and resulting tuple has max 10 elements, so if we add new field to QueryTuple it would have 11 elements which is bad :(

**Comment by appetrosyan** [comment]
Is this PR still relevant? @baziorek @iceseer

**Comment by baziorek** [comment]
> Is this PR still relevant? @baziorek @iceseer

@appetrosyan @Pawlak00 Could You please tell us if the PR is ready, or is to close?

**Pull request closed without merging**